lib/hotbar.js

```diff
@@ -19,40 +44,112 @@ class HotBar extends LitElement {
       this.reloadHotbarSelected(newSlot)
     })
 
-    this.bot.inventory.on('updateSlot', (slot, oldItem, newItem) => {
+    this.bot.inventory.on('updateSlot', async (slot, oldItem, newItem) => {
       if (slot >= this.bot.inventory.hotbarStart + 9) return
       if (slot < this.bot.inventory.hotbarStart) return
 
       const sprite = newItem ? invsprite[newItem.name] : invsprite.air
-      const slotImage = this.shadowRoot.getElementById('hotbar-' + (slot - this.bot.inventory.hotbarStart))
-      slotImage.style['background-position-x'] = `-${sprite.x * 2}px`
-      slotImage.style['background-position-y'] = `-${sprite.y * 2}px`
-      slotImage.innerHTML = newItem?.count > 1 ? newItem.count : ''
+      const slotImage = this.shadowRoot.getElementById('hotbar-' + (slot - this.bot.inventory.hotbarStart)).firstElementChild
+
+      const icon = await getItemIcon(sprite)
+      slotImage.innerHTML = ''
+      slotImage.appendChild(await icon)
+
+      slotImage.nextElementSibling.innerHTML = newItem?.count > 1 ? newItem.count : ''
     })
   }
 
   async reloadHotbar () {
     for (let i = 0; i < 9; i++) {
       const item = this.bot.inventory.slots[this.bot.inventory.hotbarStart + i]
       const sprite = item ? invsprite[item.name] : invsprite.air
-      const slotImage = this.shadowRoot.getElementById('hotbar-' + i)
-      slotImage.style['background-position-x'] = `-${sprite.x * 2}px`
-      slotImage.style['background-position-y'] = `-${sprite.y * 2}px`
-      slotImage.innerHTML = item?.count > 1 ? item.count : ''
+      const slotImage = this.shadowRoot.getElementById('hotbar-' + i).firstElementChild
+
+      const icon = await getItemIcon(sprite)
+      slotImage.innerHTML = ''
+      slotImage.appendChild(await icon)
+
+      slotImage.nextElementSibling.innerHTML = item?.count > 1 ? item.count : ''
     }
   }
 
   async reloadHotbarSelected (slot) {
     const item = this.bot.inventory.slots[this.bot.inventory.hotbarStart + slot]
-    const planned = (20 * 4 * slot) + 'px'
-    this.shadowRoot.getElementById('hotbar-highlight').style.marginLeft = planned
+
+    const hotbarSlots = this.shadowRoot.querySelectorAll('.hotbar-slot')
+    for (let i = 0; i < hotbarSlots.length; i++) {
+      const hotbarSlot = hotbarSlots[i]
+
+      hotbarSlot.classList.remove('selected')
+      if (slot === i) hotbarSlot.classList.add('selected')
+    }
+
```

Comment:
What's the reason to replace this 2 line code by this 6 line code that does the same thing but manipulate more objects ?

Reply:
I don't see any problem.
Also this way is more htmly. '-'

Reply:
The problem is to keep the code short and clean.

Reply:
How does this make the code not clean?

Reply:
The function was making 1 operation and now its doing 10 operations to do the same thing. It complexify the code and create more opportunity for mistakes (before it was not possible by construction to have multiple slot selected, now there is a possibility so it have to be enforced by code). In general its a good programming practice to have things that cannot fail by design instead of adding more logic to handle edge cases. Also shorter code is easier to read and understand especially if you add more flow control.